### PR TITLE
Feature/integration manager

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes https://github.com/vector-im/riot-android/issues/3374
### Pull Request Checklist

* [ ] Pull request is based on the develop branch
* [ ] Pull request updates [CHANGES.rst](https://github.com/matrix-org/matrix-android-sdk/blob/develop/CHANGES.rst)
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

File: matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnown.kt [hunk 1]
```diff
@@ -39,4 +50,35 @@ class WellKnown {
...
+    /**
+     * Returns the list of integration managers proposed
+     */
+    fun getIntegrationManagers(): List<WellKnownManagerConfig> {
+        val managers = ArrayList<WellKnownManagerConfig>()
+        integrations?.get("managers")?.let {
+            (it as? ArrayList<*>)?.let { configs ->
+                configs.forEach { config ->
+                    (config as? Map<*, *>)?.let { map ->
+                        val apiUrl = map["api_url"] as? String
```
Comment (bmarty) [inline matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnown.kt]: `takeIf { it.startsWith("https://") }` can be more elegant than the if condition

File: matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnownManagerConfig.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (bmarty) [inline matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnownManagerConfig.kt]: copyright

File: matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManager.kt [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+ *  The integration manager listen to account data, and can notify observer for changes.
+ *
+ *  The wellknown is refreshed at each application fresh start
+ *
+ */
+class IntegrationManager(val mxSession: MXSession, val context: Context) {
+
+
+    private val PREFS_IM = "IntegrationManager.Storage"
+    private val WELLKNOWN_KEY = "WellKnown"
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManager.kt]: Should be const in the companion

File: matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManagerConfig.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package org.matrix.androidsdk.features.integrationmanager
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManagerConfig.kt]: Copyright please

Comment (BillCarsonFr) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManagerConfig.kt]: done

File: matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManager.kt (hunk 3 above)
Comment (BillCarsonFr) [reply matrix-sdk/src/main/java/org/matrix/androidsdk/features/integrationmanager/IntegrationManager.kt]: done

File: matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnownManagerConfig.kt (hunk 2 above)
Comment (BillCarsonFr) [reply matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnownManagerConfig.kt]: done

File: matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnown.kt (hunk 1 above)
Comment (BillCarsonFr) [reply matrix-sdk-core/src/main/java/org/matrix/androidsdk/rest/model/WellKnown.kt]: will update in following PR

File: CHANGES.rst [hunk 5]
```diff
@@ -5,7 +5,7 @@ Features:
  -
 
 Improvements:
- -
+ - Integration Manager preferences are now manage by SDK
```
[the remark this answers is not in the document]
Comment (BillCarsonFr) [reply CHANGES.rst]: Done

**Pull request merged**